widgetssdk/src/main/java/com/glia/widgets/view/unifiedui/exstensions/MergeExtensions.kt

```diff
@@ -0,0 +1,28 @@
+package com.glia.widgets.view.unifiedui.exstensions
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.primaryConstructor
+
+private fun <T> mergeProperties(property: KProperty1<out T, Any?>, current: T, other: T): Any? {
+    val leftValue = property.getter.call(current)
+    val rightValue = property.getter.call(other)
+    return rightValue?.let { leftValue?.merge(it) } ?: rightValue ?: leftValue
+}
+
+infix fun <T : Any?> T.merge(other: T): T {
```

Comment:
Should we allow this method for any classes? Maybe better apply it for some `interface Mergeble` type?